From f257875751a70b62639387e44abc6c64715aa317 Mon Sep 17 00:00:00 2001
From: Tom Adams <tom@holizz.com>
Date: Sat, 3 Jan 2009 16:50:21 +0000
Subject: [PATCH] don't change mode to the current mode (allows toggling between two modes)

---
 src/joy2chord.cpp |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/src/joy2chord.cpp b/src/joy2chord.cpp
index 6833beb..63be71e 100644
--- a/src/joy2chord.cpp
+++ b/src/joy2chord.cpp
@@ -1106,9 +1106,10 @@ void joy2chord::process_events(js_event js)
 						}	*/
 					}
 				}				
+				int old_mode = mode;
 				for (int mode_loop = 0; mode_loop < MAX_MODES; mode_loop++)
 				{
-					if( mode_code[mode_loop] == button_code)
+					if(mode_code[mode_loop] == button_code && mode_loop != old_mode)
 					{
 						if (verbose)
 						{
-- 
1.6.0.6

